src/theory/arith/nl/transcendental/sine_solver.cpp

```diff
@@ -54,32 +54,134 @@ inline Node mkValidPhase(TNode a, TNode pi)
 SineSolver::SineSolver(Env& env, TranscendentalState* tstate)
     : EnvObj(env), d_data(tstate)
 {
+  NodeManager* nm = NodeManager::currentNM();
+  Node zero = nm->mkConstReal(Rational(0));
+  Node one = nm->mkConstReal(Rational(1));
+  Node negOne = nm->mkConstReal(Rational(-1));
+  d_pi = nm->mkNullaryOperator(nm->realType(), Kind::PI);
+  Node pi_2 = rewrite(
+      nm->mkNode(Kind::MULT, d_pi, nm->mkConstReal(Rational(1) / Rational(2))));
+  Node pi_neg_2 = rewrite(nm->mkNode(
+      Kind::MULT, d_pi, nm->mkConstReal(Rational(-1) / Rational(2))));
+  d_neg_pi = rewrite(nm->mkNode(Kind::MULT, d_pi, negOne));
+  d_mpoints.push_back(d_pi);
+  d_mpointsSine.push_back(zero);
+  d_mpoints.push_back(pi_2);
+  d_mpointsSine.push_back(one);
+  d_mpoints.push_back(zero);
+  d_mpointsSine.push_back(zero);
+  d_mpoints.push_back(pi_neg_2);
+  d_mpointsSine.push_back(negOne);
+  d_mpoints.push_back(d_neg_pi);
+  d_mpointsSine.push_back(zero);
 }
 
 SineSolver::~SineSolver() {}
 
+void SineSolver::doReductions()
+{
+  NodeManager* nm = NodeManager::currentNM();
+  std::map<Kind, std::vector<Node> >::iterator it =
+      d_data->d_funcMap.find(kind::SINE);
+  if (it == d_data->d_funcMap.end())
+  {
+    return;
+  }
+  std::vector<Node> mpvs;
+  for (const Node& m : d_mpoints)
+  {
+    mpvs.push_back(d_data->d_model.computeAbstractModelValue(m));
+  }
+  std::map<Node, Node> valForSym;
+  std::map<Node, Node>::iterator itv;
+  std::vector<Node> nreduced;
+  for (const Node& tf : it->second)
+  {
+    Node mva = d_data->d_model.computeAbstractModelValue(tf[0]);
+    Node mv = d_data->d_model.computeAbstractModelValue(tf);
+    Node mvaNeg = nm->mkConstReal(-mva.getConst<Rational>());
+    itv = valForSym.find(mvaNeg);
+    bool reduced = false;
+    if (itv != valForSym.end())
+    {
+      Node mvs = d_data->d_model.computeAbstractModelValue(itv->second);
+      if (mvs.getConst<Rational>() != -mv.getConst<Rational>())
+      {
+        Node lem =
+            nm->mkNode(kind::IMPLIES,
+                       tf[0].eqNode(nm->mkNode(kind::NEG, itv->second[0])),
+                       tf.eqNode(nm->mkNode(kind::NEG, itv->second)));
+        d_data->d_im.addPendingLemma(
+            lem, InferenceId::ARITH_NL_T_SINE_SYMM, nullptr);
+      }
+      // we do not consider it reduced currently, since we require setting
+      // approximate bounds for it, alternatively we could carry the negation
+      // of the approximation in the transcendental solver
```

Comment:
I would prefer to `continue;` right here, simplifying the structure of the loop body.

Reply:
We require falling through so that it is added to `nreduced` at the bottom of the loop.